src/IssueViz.Security.UnitTests/Taint/TaintList/TaintIssuesControlViewModelTests.cs

```diff
@@ -151,17 +226,80 @@ public void NavigateCommand_Execute_LocationNavigated()
             locationNavigator.VerifyNoOtherCalls();
         }
 
-        private static TaintIssuesControlViewModel CreateTestSubject(ObservableCollection<IAnalysisIssueVisualization> originalCollection = null,
+        [TestMethod]
+        public void ActiveDocumentChanged_NoActiveDocument_NoIssuesDisplayed()
+        {
+            var storeCollection = new ObservableCollection<IAnalysisIssueVisualization> { CreateIssueViz() };
+
+            var activeDocumentTracker = new Mock<IActiveDocumentTracker>();
+
+            var testSubject = CreateTestSubject(storeCollection, activeDocumentTracker: activeDocumentTracker.Object);
+
+            activeDocumentTracker.Raise(x=> x.OnDocumentFocused += null, new DocumentFocusedEventArgs(null));
+
+            testSubject.Issues.Count.Should().Be(1);
+
+            var view = CollectionViewSource.GetDefaultView(testSubject.Issues);
+            view.Filter.Should().NotBeNull();
+
+            var filteredItems = testSubject.Issues.Where(x => view.Filter(x));
+            filteredItems.Count().Should().Be(0);
+        }
+
+        [TestMethod]
+        public void ActiveDocumentChanged_ActiveDocumentExists_IssuesFilteredForActiveFilePath()
+        {
+            var issueViz1 = CreateIssueViz("test1.cpp");
+            var issueViz2 = CreateIssueViz("test2.cpp");
+            var storeCollection = new ObservableCollection<IAnalysisIssueVisualization> { issueViz1, issueViz2 };
+
+            var activeDocumentTracker = new Mock<IActiveDocumentTracker>();
+
+            var testSubject = CreateTestSubject(storeCollection, activeDocumentTracker: activeDocumentTracker.Object);
+
+            var activeDocument = new Mock<ITextDocument>();
+            activeDocument.Setup(x => x.FilePath).Returns("test2.cpp");
+
+            activeDocumentTracker.Raise(x => x.OnDocumentFocused += null, new DocumentFocusedEventArgs(activeDocument.Object));
```

Comment:
Minor: the tests that handle the raised a doc focused event would be more readable if there was a helper method that created the `ActiveDocument` and raised the event e.g.
`RaiseDocFocusedEvent(Mock<IActiveDocumentTracker> mock, string activeDocName)`
... where `activeDocName` could be null.

Reply:
It's not the document name that is null, it's the document itself.
So `activeDocName` would have a double meaning - that if it's null, it means the document itself is null (and not just the name).
It'd have to be something like:
```
void RaiseDocFocusedEvent(Mock<IActiveDocumentTracker> mock, string activeDocName) 
{
    ITextDocument document = null;
    if (activeDocName != null) 
     {
            var documentMock = new Mock<ITextDocument>();
            documentMock.Setup(x => x.FilePath).Returns(activeDocName);
            document = documentMock.Object;
     }
     ...
}
```
and that's a bit confusing IMO.

Reply:
Yes, I realise that, and that's exactly the sort of thing I'd want to hide to raise the level of abstraction at which the main test can be read.